lightning_server.cc

```diff
@@ -35,30 +33,18 @@ void LightningServer::start() {
 
   for (;;) {
     // Accept connection request
-    boost::asio::ip::tcp::socket socket(io_service_);
-    acceptor_.accept(socket);
+    boost::shared_ptr<boost::asio::ip::tcp::socket>
+        socket(new boost::asio::ip::tcp::socket(io_service_));
+    acceptor_.accept(*socket);
 
-    // Read in request
-    char request_buffer[MAX_REQ_SIZE];
-    boost::system::error_code ec;
-    std::size_t bytes_read = socket.read_some(boost::asio::buffer(request_buffer), ec);
-    std::cout << "~~~~~~~~~~Request~~~~~~~~~~\n" << request_buffer << std::endl;
+    // Read in request and construct echo response in EchoRequestHandler
+    EchoRequestHandler* erh = new EchoRequestHandler(socket);
```

Comment:
In general, we want to use [`unique_ptr`](https://mbevin.wordpress.com/2012/11/18/smart-pointers/) whenever we need to do dynamic allocation.